Source/CarthageKit/Git.swift

```diff
@@ -524,21 +571,33 @@ public func resolveReferenceInRepository(_ repositoryFileURL: URL, _ reference:
 	return ensureDirectoryExistsAtURL(repositoryFileURL)
 		.then(launchGitTask([ "rev-parse", "\(reference)^{object}" ], repositoryFileURL: repositoryFileURL))
 		.map { string in string.trimmingCharacters(in: .whitespacesAndNewlines) }
-		.mapError { error in CarthageError.repositoryCheckoutFailed(workingDirectoryURL: repositoryFileURL, reason: "No object named \"\(reference)\" exists", underlyingError: error as NSError) }
+		.mapError { error in
+			CarthageError.repositoryCheckoutFailed(
```

Comment:
I have now tried to both fix the missing `return` in multiline closures and the existing `return` in single line closures.

For the latter I could come up with a rule that works pretty well. Namely it's this:

```yaml
  single_line_return:
    included: ".*.swift"
    regex: '\.\S+ *\{ *return [^\}]+\}'
    name: "Single Line Return"
    message: "Remove the 'return' when returning from a single line closure."
    severity: warning

```

There's only one case where it doesn't work to remove the `return` as the compiler cannot recognize the type. It's `if names.isEmpty { return .success() }` in Project.swift. But I consider this to be the exception, so I've added a SwiftLint disable there.

For the multiline closures though I couldn't not come up with a regex that works for all closures. Here's what the closest I came up with (it helped me find all cases!): `\.(map|flatMap)[^\{\n]*\{[^\n]+in *\n\s*[^r\s][^e\s][^t\s][^u\s][^r\s][^n\s][^\s]`. But I'd say that there are too many cases where this doesn't work, like here:

```swift
let orderedNodes = graph.orderedNodes.map { node -> DependencyNode in
	node.dependencies = graph.edges[node] ?? []
	return node
}
```

So, I've not added that ...